pylops_mpi/DistributedArray.py

```diff
@@ -503,6 +503,35 @@ def _allgather(self, send_buf, recv_buf=None):
             self.base_comm.Allgather(send_buf, recv_buf)
             return recv_buf
 
+    def _send(self, send_buf, dest, count=None, tag=None):
+        """ Send operation
+        """
+        if deps.nccl_enabled and getattr(self, "base_comm_nccl"):
+            if count is None:
+                # assuming sending the whole array
+                count = send_buf.size
+            nccl_send(self.base_comm_nccl, send_buf, dest, count)
+        else:
+            self.base_comm.Send(send_buf, dest, tag)
+
+    def _recv(self, recv_buf=None, source=0, count=None, tag=None):
+        """ Receive operation
+        """
+        # NCCL must be called with recv_buf. Size cannot be inferred from
+        # other arguments and thus cannot dynamically allocated
+        if deps.nccl_enabled and getattr(self, "base_comm_nccl") and recv_buf is not None:
```

Comment:
mmh what if `deps.nccl_enabled and getattr(self, "base_comm_nccl")` is True but one is by mistake not passing `recv_buf`? So far I think it will go to the MPI part.... instead I think we should raise an error